api_docs/advanced_settings.json

```diff
@@ -18,7 +18,7 @@
             "source": {
               "path": "src/plugins/advanced_settings/public/component_registry/component_registry.ts",
               "lineNumber": 28,
-              "link": "https://github.com/elastic/kibana/tree/mastersrc/plugins/advanced_settings/public/component_registry/component_registry.ts#L28"
```

Comment:
I feel like we have the relative path, line number, and branch information when we're rendering the docs, maybe we should just drop the link property and produce it from the raw data in the `<DocDefinitionList />`?

Reply:
`DocDefinitionList` component will eventually be used for APIs from other repos, so I'd need to pass over the repo link too. At that point, it seems like I might as well just pass the link.

Reply:
It's still super redundant info that causes twice the changes lines in every PR when line numbers change or the path name changes and requires that we update all the json files in the repo... Am I underestimating how hard this is to implement? I'd be happy to help if I can

Reply:
I think I wasn't understanding what you are asking. I'm totally fine with removing the duplicate information (done!), I'm doing it by getting rid of the parts and keeping just the `link`.  DocDefinitionList component can work as it has (it never needed that extra information, it was just accidentally included in the JSON)

Reply:
Actually I will need to update the doc definition list to work off source, not source.link, but yea, we are on the same page! :)